File: oim-jython/find_users.py
from xlclient import *
from java.util import *

import sys

if __name__ == '__main__':
    xlclient = XLAPIClient('xelsysadm', 'xelsysadm')
    usrIntf = xlclient.usrIntf

    usrMap = HashMap()
    usrMap.put('Users.User ID', 'X*')
    usrMap.put('Users.Role', 'Full-Time')
    usrMap.put('Users.Xellerate Type', 'End-User')

    rs = usrIntf.findUsersFiltered(usrMap, ['Users.Key', 'Users.First Name'])
    xlclient.printRS(rs)

    xlclient.close()
    sys.exit(0)

File: oim-jruby/find_users.rb
require 'java'
require 'xlclient'

include_class('java.lang.Exception') {|package,name| "J#{name}" }
include_class('java.lang.String') {|package,name| "J#{name}" }
include_class 'java.lang.System' 
include_class 'java.util.HashMap'
include_class('Thor.API.tcUtilityFactory') {|package,name| "OIM#{name}"}


xlclient = XLAPIClient.new
xlclient.defaultLogin

usrIntf = xlclient.getUtility('usr')

usrMap = HashMap.new({
    'Users.User ID' => 'X*'
})

filter = ['Users.Key', 'Users.User ID'].to_java(:String)

rs = usrIntf.findUsersFiltered(usrMap, filter)
xlclient.printRS(rs)

xlclient.close
System.exit 0
